addons/website_sale/models/website.py

```diff
@@ -246,7 +246,8 @@ def _get_pl_partner_order(
 
         # no GeoIP or no pricelist for this country
         if not pricelists:
-            pricelists = pricelists.browse(website_pricelist_ids).filtered(check_pricelist)
+            pricelists = pricelists.sudo().browse(website_pricelist_ids).filtered(
+                lambda pl: check_pricelist(pl) and not(pl['country_group_ids'] and country_code))
```

Comment:
```suggestion
                lambda pl: check_pricelist(pl) and not (country_code and pl.country_group_ids))
```
Since we're handling records here, it's confusing to use dict access to fetch the field value.
Also, if country_code is not set, we could avoid fetching the field value by checking that part first ;).